cpp/src/hash/murmur_hash.cu

```diff
@@ -29,26 +29,28 @@ namespace cudf {
 namespace detail {
 
 std::unique_ptr<column> murmur_hash3_32(table_view const& input,
+                                        uint32_t seed,
```

Comment:
I can't recall immediately but the `seed` may also be of type `hash_value_type`. Need to check this.

Reply:
I think `seed` should be `hash_value_type` (or `result_type`) but not sure if there is a particular reason to use `uint32_t` everywhere instead. Also, I just noticed there are a number of mixed uses of `result_type` and `hash_value_type` in the murmur hash implementaion (e.g. [here](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/include/cudf/detail/utilities/hash_functions.cuh#L290)). Just for my own education, are they intentional?

Reply:
@PointKernel I am planning to refactor all of that. I agree it’s awkward and some of it can be unified.